python/ray/serve/_private/deployment_state.py

```diff
@@ -2805,11 +2816,20 @@ def update(self) -> bool:
         upscales = {}
         downscales = {}
 
+        # NOTE(zcin): If the deployment status is healthy, it should mean
+        # that the number of running replicas at target version is at
+        # the target number. Adding an extra check however to be defensive.
+        in_steady_state = all(
```

Comment:
`in_steady_state` is not very descriptive, it's quite unclear what it means. I would suggest either:

- "calling it what it is": `all_deployments_healthy`
- "telling me what it does": `allow_active_compaction`

Slight preference for the latter because in the future we may update thie condition but the logic within `_migrate_replicas_on_draining_nodes` will stay the same.

Reply:
Makes sense, I've changed it to `allow_active_compaction`!